sampling/batch_creator.py

```diff
@@ -1,14 +1,44 @@
-class BatchCreator():
+from time import sleep
 
-  def __init__(self, data_store, batch_size):
-    self.data_store = data_store
-    self.batch_size = batch_size
+from torch.multiprocessing import Process
+import numpy as np
 
-  def create_batches(self):
+
+class BatchCreator(Process):
     """
-      Keep creating batches in the background
+    Responsible for creating batches from the created samples
     """
-    # Read the samples on memory (data_store.samples) and generate batches.
-    # Place the batches in data_store.batches.
-    # Pause if data_store.batches is full (or if data_store.samples empty)
-    pass
\ No newline at end of file
+
+    def __init__(self, data_store, event):
+        super(BatchCreator, self).__init__()
+        self.ds = data_store
+        self.batch_size = self.ds.batch_size
+        self.batches = self.ds.batches
+        self.max_batches = self.ds.max_batches
+        self.offset = 0
+        self.stop_batch_creator = event
+        self.is_stop = False
+
+    def run(self):
+        """
+        Keep creating the batches in the background
+        """
+        while not self.stop_batch_creator.is_set():
+            if self.ds.samples.empty():
+                print("Waiting for sample creator to create a sample")
+                continue
+            elif self.batches.full():
+                print("Waiting for client to take a batch from batches array")
+                continue
+            else:
+                i = 0
+                curr_sample = []
+                if not curr_sample or self.offset == self.ds.sample_size:
```

Comment:
You are setting curr_sample to [] right above this if statement. This if condition will always hold true.